src/material/schematics/ng-update/migrations/theme-base-v17/migration.ts

```diff
@@ -0,0 +1,316 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/** Preamble to insert before the missing mixins. */
+const MISSING_MIXIN_PREAMBLE_LINES = `\
+// The following mixins include base theme styles that are only needed once per application. These
+// theme styles do not depend on the color, typography, or density settings in your theme. However,
+// these styles may differ depending on the theme's design system. Currently all themes use the
+// Material 2 design system, but in the future it may be possible to create theme based on other
+// design systems, such as Material 3.
+//
+// Please note: you do not need to include the 'base' mixins here, if you include the corresponding
+// 'theme' mixin elsewhere in your app. The full 'theme' mixins already include the base styles.
+//
+// TODO: Please move these @include statements to the preferred place in your Sass, and pass your
+//  real theme rather than the dummy theme. This will ensure the correct values for your app are
+//  used.
+$dummy-theme: ();`.split('\n');
```

Comment:
Might be best to rename this as `empty-map` or `empty-theme`

Reply:
I went with your suggestion to actually have it not compile until the user takes some action. The added blocks now look like this:

```scss
// The following mixins include base theme styles that are only needed once per application. These
// theme styles do not depend on the color, typography, or density settings in your theme. However,
// these styles may differ depending on the theme's design system. Currently all themes use the
// Material 2 design system, but in the future it may be possible to create theme based on other
// design systems, such as Material 3.
//
// Please note: you do not need to include the 'base' mixins, if you include the corresponding
// 'theme' mixin elsewhere in your Sass. The full 'theme' mixins already include the base styles.
//
// TODO(v17): Please move these @include statements to the preferred place in your Sass, and pass
// your theme to them. This will ensure the correct values for your app are included.
@include mat.button-base(/* TODO(v17): pass $your-theme here */);
```

Reply:
The final message printed after the migration finishes is also changed, and now reads:
```
Added 1 mixin to 1 file. Please search for, and address, any "TODO(v17)" comments.
```